File: Output Algorithms/Custom/RGB/Pico.cpp
const long animation[ANIMATION_LENGTH][FRAME_LENGTH] = {...};

void plot(int x, int y, int r, int g, int b)
{
    // Insert your code here
}

void clear()
{
    // Insert your code here
}

void setup()
{
    // Insert your code here
}

int main()
{
    while (true)
    {
        for (int i = 0; i < ANIMATION_LENGTH; ++i)
        {
            int bits[width * height][3];
            for (int j = 0; j < width * height; ++j)
            {
                bits[j][0] = animation[i][j] >> 16 & 255;
                bits[j][1] = animation[i][j] >> 8 & 255;
                bits[j][2] = animation[i][j] & 255;
            }
            for (int j = 0; j < width * height; ++j)
            {
                int x = j % width;
                int y = j / width;
                plot(x, y, bits[j][0], bits[j][1], bits[j][2]);
            }
            delay(1000 / fps);
            clear();
        }
    }
    return 0;
}

File: Output Algorithms/Custom/Variable Brightness/Pico.py
import time

animation = [...]

def plot(x, y, value):
    # Insert your code here

def clear():
    # Insert your code here

while True:
    if width * height % 4 == 0:
        for i in range(len(animation)):
            bits = []
            for j in range(width * height):
                for k in range(4):
                    bits.append(animation[i][j] >> (3 - k) * 8 & 255)
            for j in range(width * height):
                x = j % width
                y = j // width
                plot(x, y, bits[j])
            time.sleep(1000 / fps)
            clear()
    else:
        for i in range(len(animation)):
            bits = []
            for j in range(width * height % 4):
                bits.append(animation[i][0] >> width * height % 4 - 1 - j & 1)
            for j in range((width * height - width * height % 4) / 4):
                for k in range(4):
                    bits.append(animation[i][j + 1] >> (3 - k) * 8 & 255)
            for j in range(width * height):
                x = j % width
                y = j // width
                plot(x, y, bits[j])
            time.sleep(1000 / fps)
            clear()